src/server/saved_objects/service/create_saved_objects_service.js

```diff
@@ -71,7 +77,7 @@ export function createSavedObjectsService(server) {
       const { callWithRequest } = server.plugins.elasticsearch.getCluster('admin');
       const callCluster = (...args) => callWithRequest(request, ...args);
 
-      const repository = new SavedObjectsRepository({
+      const repository = repositoryProvider.getRepository({
         index,
```

Comment:
We shouldn't need to pass the `index`/`mappings`/`onBeforeWrite` here anymore, right?

Reply:
Right, we don't need to pass them through anymore...I had a justification for leaving it this way, but I forgot what it was, so I'll fix this unless I remember what I was thinking